File: src/main/java/io/femo/http/drivers/server/HttpMiddlewareHandle.java
package io.femo.http.drivers.server;

import io.femo.http.*;

/**
 * Created by felix on 4/26/16.
 */
public class HttpMiddlewareHandle implements HttpHandle {

    private String path;
    private HttpMiddleware httpMiddleware;

    @Override
    public boolean matches(HttpRequest request) {
        if(path != null) {
            if(!request.path().equals(path)) {
                return false;
            }
        }
        return true;
    }

    @Override
    public boolean handle(HttpRequest request, HttpResponse response) throws HttpHandleException {
        httpMiddleware.handle(request, response);
        return false;
    }

    public String getPath() {
        return path;
    }

    public void setPath(String path) {
        this.path = path;
    }

    public HttpMiddleware getHttpMiddleware() {
        return httpMiddleware;
    }

    public void setHttpMiddleware(HttpMiddleware httpMiddleware) {
        this.httpMiddleware = httpMiddleware;
    }
}

File: src/main/java/org/web25/http/drivers/server/HttpMiddlewareHandle.kt
package org.web25.http.drivers.server

import org.web25.http.HttpRequest
import org.web25.http.exceptions.HttpHandleException
import org.web25.http.server.HttpMiddleware
import org.web25.http.server.HttpRoutable.Companion.joinPaths
import org.web25.http.server.IncomingHttpRequest
import org.web25.http.server.OutgoingHttpResponse


/**
 * Created by felix on 4/26/16.
 */
class HttpMiddlewareHandle : HttpHandle {

    var path: String? = null
    var httpMiddleware: HttpMiddleware? = null

    override fun matches(request: HttpRequest): Boolean {
        if (path != null) {
            if (request.path() != path) {
                return false
            }
        }
        return true
    }

    @Throws(HttpHandleException::class)
    override fun handle(request: IncomingHttpRequest, response: OutgoingHttpResponse): Boolean {
        httpMiddleware!!(request, response)
        return false
    }

    override fun parentPath(path: String) {
        if (this.path != null)
            this.path = joinPaths(path, this.path)
    }

    override fun prependPath(path: String) {
        this.path = joinPaths(path, this.path)
    }
}
